app/models/user.rb

```diff
@@ -13,4 +13,20 @@ def self.all_except(user)
     where.not(id:  ids += user.friends.ids)
   end
 
+  def self.to_csv
+    attributes = %w{ name state email total_Friends}
+
+    CSV.generate(headers: true) do |csv|
+      csv << attributes
+
+      all.each do |user|
+        csv << attributes.map{ |attr| user.send(attr) }
+      end
+    end
+  end
+
+  def total_Friends
+    self.friends.count
```

Comment:
No need to use `self` here. The convention to use self is when you want to assign some value to object
e.g., `self.first_name = "John"` in other we try to avoid using it explicitly